Dockerfile

```diff
@@ -2,14 +2,12 @@ FROM particle/buildpack-base:0.3.8
 
 ARG GCC_ARM_URL
 ARG GCC_ARM_VERSION
-ARG CMAKE_URL
 
 RUN dpkg --add-architecture i386 \
   && apt-get update -q && apt-get install -qy make isomd5sum bzip2 vim-common libarchive-zip-perl libc6:i386 \
-  && curl -o /tmp/cmake_install.sh -sSL ${CMAKE_URL} \
-  && chmod +x /tmp/cmake_install.sh \
-  && /tmp/cmake_install.sh --skip-license --prefix=/usr/local \
```

Comment:
`cmake` is bundled with ARM GCC? I think Sergei added it on purpose: https://github.com/particle-iot/buildpack-hal/commit/275088d3de537d8a9b3466a4284375e17d75e964#diff-3254677a7917c6c01f55212f86c57fbf

Reply:
@sergeuz Is CMake a requirement? I know we are planning to migrate, but it is not required yet. My goal is to simplify the Dockerfiles, and only install things as they become necessary.

Reply:
Oh cool, I didn't realize it is actually in use in the repository. That's great!

Reply:
@sergeuz Is there a particular version you need, or can we install `cmake` using `apt-get`?

Reply:
The top-level CMakeLists.txt mentions 3.7: https://github.com/particle-iot/device-os/blob/mesh-develop/test/unit_tests/CMakeLists.txt#L1, and Ubuntu 16.04 which build packs are currently based on has an older version

Reply:
Are you specifically using features from version 3.7, or was that version arbitrarily chosen?

Reply:
I just confirmed `apt-get` installs `cmake` v3.5.1.

Reply:
It will probably work with 3.5 as well, since catch2 is the only dependency of that test suite at the moment, and the minimum cmake version supported by catch2 is 3.5. But what's the problem with installing the latest cmake?

Reply:
Two different thoughts:

1. Can we utilize CMake 3.5.1 instead of 3.7?
1. Do we _need_ Ubuntu 16.04, or can we simply choose a different base in the buildpacks?

Reply:
I won't mind switching to 18.04, but, again, what problem are we trying to solve?

Reply:
Our goal is to ensure the integrity of all libraries/binaries running in our images. It is "safer" (more trusted) to install them from `apt-get` than download and install.

Reply:
ok, I see, so you don't want to add another checksum validation for the downloaded cmake installer. If that saves time, then let's make sure the firmware tests build and run fine with 3.5 and remove the installer from the image. We can always switch to 18.04 later on if necessary

Reply:
btw, validating a file against the checksum downloaded exactly from the same site doesn't really makes things more secure